libpod/oci_internal_linux.go

```diff
@@ -130,10 +130,16 @@ func (r *OCIRuntime) createOCIContainer(ctr *Container, restoreOptions *Containe
 	}
 
 	if ctr.config.NetMode.IsSlirp4netns() {
-		ctr.rootlessSlirpSyncR, ctr.rootlessSlirpSyncW, err = os.Pipe()
-		if err != nil {
-			return errors.Wrapf(err, "failed to create rootless network sync pipe")
+		if ctr.config.PostConfigureNetNS {
+			ctr.rootlessSlirpSyncR, ctr.rootlessSlirpSyncW, err = os.Pipe()
+			if err != nil {
+				return errors.Wrapf(err, "failed to create rootless network sync pipe")
+			}
+		} else {
+			defer errorhandling.CloseQuiet(ctr.rootlessSlirpSyncR)
```

Comment:
So we have two places where we initialize the pipes for slirp, and two places where we do a `defer close` on the pipes, and those places are not the same... Why is this? It doesn't seem to make sense.

Reply:
Right, this took me a minute to figure out. Slirp4netns is connected to conmon via this pipe; that's how it knows to die (when the container + conmon stops, the pipe closes at the conmon end, and slirp4netns stops). So, one end of the pipe must be in conmon command, and the other in the slirp4netns command. Depending on whether the new default logic is ran (setup network -> create container) or the PostConfigureNetNS logic (create container -> setup network), the timing of creating and closing the pipe changes.

I looked a little bit into creating the pipe somewhere before both operations, and then closing after both, but didn't see a good place in the code that popped out. I'm definitely open to suggestions!